src/OrchardCore.Modules/OrchardCore.Search/Controllers/SearchController.cs

```diff
@@ -142,6 +150,15 @@ public async Task<IActionResult> Search(SearchViewModel viewModel, PagerSlimPara
                 .Where(x => x.ContentItemId.IsIn(searchResult.ContentItemIds) && x.Published);
         }
 
+        var searchContext = new SearchContext
+        {
+            Index = viewModel.Index,
+            Terms = viewModel.Terms,
+            Hits = await query.CountAsync(),
+        };
+
+        await _searchHandlers.InvokeAsync(async handler => await handler.SearchedAsync(searchContext), _logger);
```

Comment:
@hishamco I saw that you suggested the opposite but here we don't need the async/await. When an async method / func returns another async method, we only need to use async/await if before returning we already needed to call another async method and then already needed to use the await keyword. Both are right but returning the Task directly is less resource consuming, and that's what we are using everywhere else, so I would first suggest.

    await _searchHandlers.InvokeAsync(handler => handler.SearchedAsync(searchContext), _logger);
 
But for optimization `InvokeAsync()` has different overloads using different typed delegates with 0, 1, ... up to 5 generic parameters, then from memory (need to be tried) actual params would need to be passed like this.

    await _searchHandlers.InvokeAsync((handler, context) =>
        handler.SearchedAsync(context), searchContext, _logger);

Reply:
> Both are right but returning the Task directly is less resource consuming

I didn't measure this before, but I TRUST you :)

Reply:
Okay, but mainly the params should be passed differently, see how `InvokeAsync()` is used in other places.

Reply:
@jtkech yes I changed it and passed the context as a second parameter. Please let me know if you need any other changes or approve.